packages/test-runner/src/reporter/reportTestsErrors.ts

```diff
@@ -75,15 +75,21 @@ export function reportTestsErrors(
     if (session.testResults) {
       const flattenedTests = getFlattenedTestResults(session.testResults);
       for (const test of flattenedTests) {
-        if (test.error) {
+        if (!test.passed && !test.skipped) {
           let testErrorsForBrowser = testErrorsPerBrowser.get(test.name);
           if (!testErrorsForBrowser) {
             testErrorsForBrowser = new Map<string, TestResultError>();
             testErrorsPerBrowser.set(test.name, testErrorsForBrowser);
           }
-          if (test.error) {
-            testErrorsForBrowser.set(session.browser.name, test.error);
-          }
+
+          test.error = test.error || { 
+            name: 'Unknown error', 
+            message: 'Test failed without any errors. One reason could be that you have code outside of your test.',
+            expected: "",
+            actual: "",
+            stack: "",
```

Comment:
It would be great to preserve the stack if possible. Is that something you think we can do here? I'd like to give the user more information.

Reply:
Nope sadly not - we can just hint which test failed. We only get `{ name: 'failing test', passed: false }` as context.

Compared to actually getting the error that we can forward as it normally would be the case, e.g.
```js
{
  duration: 1,
  error: {
    message: 'some error within a test',
    name: 'Error',
    stack: '  at qy.<anonymous> (test/blocks/global-navigation/foo.test.js:7:11)'
  },
  name: 'minimal reproducible example > this test works',
  passed: false
}
```